### PR TITLE
test: periph_timer: check if argument is passed.

### PR DESCRIPTION
**Pull request by basilfx**
In my timer driver implementation I forgot to pass `arg` to `timer_isr_ctx_t`. I noticed this when experimenting with #7919.

This PR modifies the `periph_timer` test to actually check if the argument is passed-on. I use an unique combination of cookie value + timer number + channel number to actually make sure that the right timer and channel fired (the cookie should prevent comparing to (un)initialized memory).

### CODE REVIEW

**Comment by basilfx** [comment]
Ping @haukepetersen.

File: tests/periph_timer/main.c [hunk 1]
```diff
@@ -34,15 +34,17 @@
 #define MAX_CHANNELS        (10U)
 #define TIM_SPEED           (1000000ul) /* try to run with 1MHz */
 #define CHAN_OFFSET         (5000U)     /* fire every 5ms */
+#define COOKIE              (100U)      /* for checking if arg is passed */
 
 static volatile int fired;
 static volatile uint32_t sw_count;
 static volatile uint32_t timeouts[MAX_CHANNELS];
+static volatile int args[MAX_CHANNELS];
```
Comment (haukepetersen) [inline tests/periph_timer/main.c]: this needs to be `unsinged` (or you have to do some casting below), otherwise gcc complains..

Comment (basilfx) [reply tests/periph_timer/main.c]: Changed it to uint32_t, and the initial value to UINT32_MAX.

**Comment by basilfx** [comment]
It's green. Commit message is wrong, but I'll squash if OK.

**Comment by haukepetersen** [comment]
ok

**Comment by basilfx** [comment]
All squashed and green, OK to merge?

**Comment by aabadie** [comment]
> OK to merge?

Yes, go ahead :)

**Pull request merged**